### PR TITLE
Fixed #29530 -- Fixed aliases ordering when chaining annotate() and filter().

### PR DESCRIPTION
**Pull request by felixxm**
Ticket [29530](https://code.djangoproject.com/ticket/29530).

### CODE REVIEW

**Comment by timgraham** [comment]
Do you think a comment would be useful? As someone not very familiar with the ORM internals, I have little idea of why that code is correct and the significance of `[-1]`.

**Comment by felixxm** [comment]
@timgraham I added comment, please take a look.

File: django/db/models/sql/query.py [hunk 1]
```diff
@@ -934,8 +934,14 @@ def join(self, join, reuse=None, reuse_with_filtered_relation=False):
                 if (reuse is None or a in reuse) and j == join
             ]
         if reuse_aliases:
-            self.ref_alias(reuse_aliases[0])
-            return reuse_aliases[0]
+            if join.table_alias in reuse_aliases:
+                reuse_alias = join.table_alias
+            else:
+                # Reuse the most recent alias of the joined table, because
```
Comment (timgraham) [inline django/db/models/sql/query.py]: Reuse the most recent alias of the joined table (a many-to-many relation may be joined multiple times).

**Comment by felixxm** [comment]
@timgraham Thanks for the review.

**Pull request merged**